File: solved.ac/CLASS2/1929.py
import sys
input = sys.stdin.readline

m, n = map(int, input().split())

arr = [True for i in range(n+1)]

for i in range(2, n+1):
    if arr[i]:
        if i >= m:
            print(i)
        num = i
        a = num
        while a + num <= n:
            a += num
            arr[a] = False

File: solved.ac/CLASS2_JS/1929.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const solution = () => {
  const [m, n] = input().split(" ").map(Number);
  const arr = Array(n + 1).fill(0);
  const ans = [];

  for (let i = 2; i < n + 1; i++) {
    if (arr[i] === 0) {
      if (i >= m) {
        ans.push(i);
      }
      for (let j = i + i; j <= n; j += i) {
        arr[j] = 1;
      }
    }
  }
  console.log(ans.join("\n"));
};

solution();
